roocs_utils/catalog_maker/catalog.py

```diff
@@ -0,0 +1,190 @@
+import glob
+import os
+import sys
+import time
+from collections import OrderedDict
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+import yaml
+
+from roocs_utils import CONFIG
+from roocs_utils.catalog_maker.utils import create_dir
+from roocs_utils.project_utils import DatasetMapper
+from roocs_utils.xarray_utils.xarray_utils import get_coord_type
+from roocs_utils.xarray_utils.xarray_utils import open_xr_dataset
+
+
+def get_time_info(ds, var_id):
+    all_times = []
+    try:
+
+        times = ds[var_id].time.values
+
+        all_times.extend(list(times))
+        ds.close()
+    except AttributeError:
+        return "undefined", "undefined"
+
+    return (
+        all_times[0].isoformat(timespec="seconds"),
+        all_times[-1].isoformat(timespec="seconds"),
+    )
+
+
+def get_coord_info(ds):
+    d = OrderedDict()
+
+    for coord_id in sorted(ds.coords):
+
+        coord = ds.coords[coord_id]
+        coord_type = get_coord_type(coord)
+
+        if coord_type == "time" or coord_type is None:
+            continue
+
+        data = coord.values
+
+        mn, mx = data.min(), data.max()
+
+        if np.isnan(mn) or np.isnan(mx):
+            mn, mx = float(coord.min()), float(coord.max())
+
+        if coord_type == "longitude":
+            if mn < -360 or mx > 360:
+                raise Exception(
+                    f"Longitude is not within expected bounds. The minimum and maximum are {mn}, {mx}"
+                )
+
+        if coord_type == "latitude":
+            if mn < -90 or mx > 90:
+                raise Exception(
+                    f"Latitude is not within expected bounds. The minimum and maximum are {mn}, {mx}"
+                )
+
+        d[f"{coord_type}"] = f"{mn:.2f} {mx:.2f}"
+
+    return d
+
+
+def get_size_data(ds):
+
+    size = ds.nbytes
+    size_gb = round(size / 1e9, 2)
+
+    return size, size_gb
+
+
+def get_files(ds_id):
+    fpaths = DatasetMapper(ds_id).files
+
+    if len(fpaths) < 1:
+        raise FileNotFoundError("No files were found for this dataset")
+
+    return fpaths
+
+
+def build_dict(ds_id, fpath, proj_dict):
+    comps = ds_id.split(".")
+    ds = open_xr_dataset(fpath)
+
+    facet_rule = proj_dict["facet_rule"]
+    facets = dict([_ for _ in zip(facet_rule, comps)])
+
+    var_id = facets.get("variable") or facets.get("variable_id")
+
+    size, size_gb = get_size_data(ds)
+    start_time, end_time = get_time_info(ds, var_id)
+    coord_d = get_coord_info(ds)
+
+    d = OrderedDict()
+
+    d["ds_id"] = ds_id
+    d["path"] = "/".join(ds_id.split(".")[1:]) + fpath.split("/")[-1]
+
+    d["size"] = size
+    d["size_gb"] = size_gb
```

Comment:
I think for the csv catalog it is enough to just have the size (bytes). Other tools using the catalog can display more nicely. Just a hint:
https://pypi.org/project/humanize/

Reply:
👍

Reply:
@ellesmith88 @cehbrecht: my thoughts on these issues:
1. I think "bbox" is preferred to lat and lon bounds
2. I would like to keep both `size` and `size_gb`.  The latter is just a human-readable aid.

Reply:
I'll change to `bbox` 😃

Reply:
>     2\. I would like to keep both `size` and `size_gb`.  The latter is just a human-readable aid.

@agstephens it does not hurt to keep `size_gb`. But I would keep the intake csv catalog only as information base. The display and evaluation can be done with pandas etc.